tool/site/client-react/src/pages/myPrescriptionsPage.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react"
+import {useContext} from "react"
+import {Label, Table} from "nhsuk-react-components"
+import {AppContext} from "../index"
+import LongRunningTask from "../components/longRunningTask"
+import {axiosInstance} from "../requests/axiosInstance"
+import PrescriptionActions from "../components/prescriptionActions"
+
+const MyPrescriptionsPage: React.FC = () => {
+  const {baseUrl} = useContext(AppContext)
+  const retrievePrescriptionsTask = () => retrievePrescriptions(baseUrl)
+
+  return (
+    <LongRunningTask<Prescriptions> task={retrievePrescriptionsTask} loadingMessage="Retrieving prescriptions.">
+      {prescriptions => {
+        if (!prescriptions.any) {
+          return (
+            <>
+              <Label isPageHeading>My Prescriptions</Label>
+              <p>You do not have any active prescriptions. Once you create or release a prescription you will see it here</p>
+            </>
+          )
+        }
+        return (
+          <>
+            <Label isPageHeading>My Prescriptions</Label>
+            <PrescriptionGroupTable
+              name="Sent Prescriptions"
+              description="Prescriptions ready to release"
+              prescriptions={prescriptions.sentPrescriptions}
+              actions={{view: true, release: true}}
+            />
+            <PrescriptionGroupTable
+              name="Released Prescriptions"
+              description="Prescriptions ready to dispense"
+              prescriptions={prescriptions.releasedPrescriptions}
+              actions={{view: true, dispense: true}}
+            />
+          </>
+        )
+      }}
+    </LongRunningTask>
+  )
+}
+
+interface Prescriptions {
+  any: Boolean
```

Comment:
think we can infer `prescriptions.any` based on the length of the other two arrays?

Reply:
can do, the logic looked horrrible in the client, so i moved it to server, if you prefer it in the client I can change it. let me know if it's a sticking point for you